plugins/traefik-vhosts/docker-args-process-deploy

```diff
@@ -0,0 +1,121 @@
+        proxy_container_http_port_candidate="$proxy_container_port"
+        proxy_host_http_port_candidate="$proxy_host_port"
+      fi
+
+      if [[ "$proxy_host_port" == "80" ]] && [[ -z "$proxy_container_http_port" ]]; then
+        proxy_container_http_port="$proxy_container_port"
+      fi
+    fi
+
+    if [[ "$proxy_scheme" == "https" ]]; then
+      is_app_listening="true"
+      if [[ -z "$proxy_container_https_port_candidate" ]]; then
+        proxy_container_https_port_candidate="$proxy_container_port"
+        proxy_host_https_port_candidate="$proxy_host_port"
+      fi
+
+      if [[ "$proxy_host_port" == "443" ]] && [[ -z "$proxy_container_https_port" ]]; then
+        proxy_container_https_port="$proxy_container_port"
+      fi
+    fi
+  done
+
+  # add the labels for traefik here
+  # any `http:80` port mapping is treated as a `http` traefik entrypoint
+  # any `https:443` port mapping is treated as a `https` traefik entrypoint
+  if [[ -n "$is_app_listening" ]]; then
+    app_domains="$(plugn trigger domains-list "$APP")"
+    if [[ -n "$app_domains" ]]; then
+      traefik_domains="$(echo "$app_domains" | xargs)"
+      traefik_domains="${traefik_domains// /\\\`,\\\`}"
+    fi
+
+    output="--label traefik.enable=true"
+    if [[ -n "$proxy_container_http_port" ]] || [[ -n "$proxy_container_http_port_candidate" ]]; then
+      if [[ -z "$proxy_container_http_port" ]]; then
+        dokku_log_warn "Warning: http:80 port mapping not found"
+        dokku_log_warn "Utilizing first http port mapping, http:$proxy_host_http_port_candidate:$proxy_container_http_port_candidate"
+        proxy_container_http_port="$proxy_container_http_port_candidate"
+      fi
+
+      output="$output --label traefik.http.services.$APP-$PROC_TYPE.loadbalancer.server.port=$proxy_container_http_port"
+      output="$output --label traefik.http.routers.$APP-$PROC_TYPE.entrypoints=http"
+      if [[ -n "$traefik_domains" ]]; then
+        output="$output --label \"traefik.http.routers.$APP-$PROC_TYPE.rule=Host(\\\`$traefik_domains\\\`)\""
+        echo "# THIS FILE IS GENERATED BY DOKKU - DO NOT EDIT, YOUR CHANGES WILL BE OVERWRITTEN" >"$app_urls_path"
+        xargs -I{} echo "http://{}" <<<"$(echo "${app_domains}" | tr ' ' '\n' | sort -u)" >>"$app_urls_path"
+      fi
+    fi
+
+    if [[ -n "$proxy_container_https_port" ]] || [[ -n "$proxy_container_https_port_candidate" ]]; then
+      if [[ -z "$proxy_container_https_port" ]]; then
+        dokku_log_warn "Warning: https:443 port mapping not found"
+        dokku_log_warn "Utilizing first https port mapping, http:$proxy_host_https_port_candidate:$proxy_container_https_port_candidate"
+        proxy_container_https_port="$proxy_container_https_port_candidate"
+      fi
+
+      output="$output --label traefik.http.services.$APP-$PROC_TYPE-https.loadbalancer.server.port=$proxy_container_https_port"
+      output="$output --label traefik.http.routers.$APP-$PROC_TYPE-https.entrypoints=https"
```

Comment:
Shouldn't we specify a `certresolver` for each service?

Like
```bash
output="$output --label traefik.http.routers.$APP-$PROC_TYPE-https.tls.certresolver=leresolver"
```

Without specifying a `certresolver` it seems like Traefik won't even try to use TLS for this route...